src/adjoint.jl

```diff
@@ -1,16 +1,53 @@
 using MacroTools
 using MacroTools: @q, combinedef
+using ChainRulesCore: AbstractZero, Zero, DoesNotExist
 
 named(arg) = isexpr(arg, :(::)) && length(arg.args) == 1 ? :($(gensym())::$(arg.args[1])) : arg
 
 typeless(x) = MacroTools.postwalk(x -> isexpr(x, :(::), :kw) ? x.args[1] : x, x)
 isvararg(x) = isexpr(x, :(::)) && namify(x.args[2]) == :Vararg
 
+function legacytype_warn()
+  # can't use logging macros as that breaks nested AD.
+  Core.println("Zygote internal use  of 'nothing', rather than `AbstractZero`, detected.  This should never occur. Please open an issue on https://github.com/FluxML/Zygote.jl/issues, including the full text of this message. \n Stacktrace:")
+  for (ii, callsite) in enumerate(stacktrace())
```

Comment:
This is a really tiny thing, but why use `ii` as a variable name here? I'm not especially opposed to it, I just don't think I've seen it before.

Reply:
No particular reason, changed to `i` which is more canonical